Step 2

JohnnyGsy opened the pull request.

JohnnyGsy committed:
books controllers added,pagy,made some buttons active

JohnnyGsy committed:
decorator spec corrected, rubocoped

JohnnyGsy committed:
bootstrap js fixed

JohnnyGsy committed:
covering with specs

kolyanjy commented on .vscode/settings.json:
```diff
@@ -0,0 +1,5 @@
+  "i18n-ally.localesPaths": [
+    "config/locales"
+  ]
+}
```
to git ignore

kolyanjy commented on Gemfile:
```diff
@@ -5,19 +5,19 @@ ruby '2.7.6'
 gem 'jbuilder', '2.11.5'
-
 # Database
+gem 'jquery-rails'
```
database group 0_o

kolyanjy commented on Gemfile:
```diff
@@ -5,19 +5,19 @@ ruby '2.7.6'
 gem 'pg', '1.3.5'
-
 # Puma web server
+gem 'popper_js'
```
pls put gems into correct groups

kolyanjy commented on Gemfile:
```diff
@@ -36,6 +36,12 @@ gem 'turbo-rails', '1.0.1	'
+gem 'net-http'
+
+# Pagination gem
+gem 'pagy', '~> 5.10'
```
👍

kolyanjy commented on app/assets/config/manifest.js:
```diff
@@ -1,4 +1,9 @@
+//= jquery
+//= jquery_ujs
+//= popper
+//= bootstrap
```
empty line

kolyanjy commented on app/assets/config/manifest.js:
```diff
@@ -1,4 +1,9 @@
 //= link_directory ../stylesheets .css
 //= link_tree ../../javascript .js
 //= link_tree ../../../vendor/javascript .js
+//= jquery
+//= jquery_ujs
+//= popper
+//= bootstrap
```
try to move it into `application.js`

JohnnyGsy replied:
Already imported to application.js but it didn't work without import into manifest.js

kolyanjy replied:
okay

kolyanjy commented on app/decorators/book_decorator.rb:
```diff
@@ -1,8 +1,13 @@
 class BookDecorator < Draper::Decorator
+  TRUNCATED_LENGTH = { description: 250 }.freeze
```
just value without hash

kolyanjy commented on app/javascript/controllers/descriprion_controller.js:
```diff
@@ -0,0 +1,27 @@
+    this.moreButtonTarget.hidden = true
+    this.longTextTarget.hidden = false
+    this.lessButtonTarget.hidden = false
+    console.log('show more')
```
remove logger

kolyanjy commented on app/javascript/controllers/descriprion_controller.js:
```diff
@@ -0,0 +1,27 @@
+    this.moreButtonTarget.hidden = false
+    this.longTextTarget.hidden = true
+    this.lessButtonTarget.hidden = true
+    console.log('show less')
```
the same

kolyanjy commented on app/javascript/application.js:
```diff
@@ -1,3 +1,7 @@
+import "jquery"
+import "jquery_ujs"
+import "popper"
+import "bootstrap"
```
empty line

kolyanjy replied:
Pls add this rule to rubocop

kolyanjy commented on app/services/sort_books_service.rb:
```diff
@@ -0,0 +1,30 @@
+
+  private
+
+  def find_books
+    @books = @category_id ? Book.where(category_id: @category_id) : Book.all
+  end
```
you can move it to initializer

kolyanjy commented on app/services/sort_books_service.rb:
```diff
@@ -0,0 +1,30 @@
+  }.freeze
+  def initialize(params)
+    @category_id = params[:category_id]
+    @filter = params[:filter]&.to_sym || BOOK_DEFAULT_FILTER
```
```ruby
@filter = params[:filter]&.to_sym || BOOK_FILTERS[:created_at_desc]
```

kolyanjy replied:
```suggestion
    @filter = params[:filter]&.to_sym || BOOK_FILTERS[:created_at_desc]
```

kolyanjy commented on app/services/sort_books_service.rb:
```diff
@@ -0,0 +1,30 @@
+class SortBooksService
```
its a query object

kolyanjy commented on config/initializers/assets.rb:
```diff
@@ -2,10 +2,9 @@
 # Add additional assets to the asset load path.
 # Rails.application.config.assets.paths << Emoji.images_path
-
+Rails.application.config.assets.precompile += %w[jquery.min.js jquery_ujs.js bootstrap.min.js popper.js]
```
change on manifest.js

kolyanjy commented on config/importmap.rb:
```diff
@@ -5,3 +5,7 @@
 pin '@hotwired/stimulus', to: 'stimulus.min.js', preload: true
 pin '@hotwired/stimulus-loading', to: 'stimulus-loading.js', preload: true
 pin_all_from 'app/javascript/controllers', under: 'controllers'
+pin 'jquery', to: 'jquery.min.js', preload: true
+pin 'jquery_ujs', to: 'jquery_ujs.js', preload: true
+pin 'popper', to: 'popper.js', preload: true
+pin 'bootstrap', to: 'bootstrap.min.js', preload: true
```
👍

kolyanjy commented on spec/feature/books/show_spec.rb:
```diff
@@ -0,0 +1,19 @@
+RSpec.describe 'books#show', type: :feature do
+  let(:book) { create(:book) }
+
```
pls add specs with increasing count of books or you can add it with specs on cart

kolyanjy commented on spec/feature/books/index_spec.rb:
```diff
@@ -0,0 +1,29 @@
+    let!(:another_category_books) do
+      BookDecorator.decorate_collection(create_list(:book, books_amount, category: another_category))
+    end
+
```
add specs for pagination

kolyanjy reviewed (changes requested):
Pls fix comment and change base branch to develop

JohnnyGsy committed:
moved sort service to queries ,fixed vies,gitignored some files

JohnnyGsy committed:
added some specs,fixed sort_books

JohnnyGsy committed:
Updated config.yml

Updated config.yml

Updated config.yml

Updated config.yml

Updated config.yml

fixing ci

fixing ci

fixing ci

fixing ci

fixing ci

fixing ci

fixing ci

fixing ci

fixing ci
